core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -71,6 +82,99 @@ private[spark] class CoarseGrainedExecutorBackend(
     }(ThreadUtils.sameThread)
   }
 
+  // Check that the executor resources at startup will satisfy the user specified task
+  // requirements (spark.task.resource.*) and that they match the executor configs
+  // specified by the user (spark.executor.resource.*) to catch mismatches between what
+  // the user requested and what resource manager gave or what the discovery script found.
+  private def checkExecResourcesMeetTaskRequirements(
```

Comment:
We can have a general function works for both driver and executor. Essentially, the request is `resourcePrefix -> count` and the resource is `resourcePrefix -> addresses`. If they don't match, we just include `resourcePrefix` in the error message, which is clear to users whether it is driver or executor.

Reply:
I don't completely follow what you are asking for here. We are comparing 3 things
spark.task.resource.* -> count with spark.executor.resource.* -> count and with the actual found in the script or pass in which is a Map[resourceName, ResourceInformation].  You can't have the resourceprefix on the type else it won't compare properly to the Map[resourceName] 

I can certainly make it more generic to handle both executor and driver and I made some code changes to go that way but I would prefer to wait til the jira that implements the Driver side to finish that to make sure we don't need anything else. This function will likely have to move somewhere anyway.